cmd/bootstrap/run/epochs.go

```diff
@@ -149,10 +152,19 @@ func GenerateRecoverTxArgsWithDKG(log zerolog.Logger,
 	internalNodesMap := make(map[flow.Identifier]struct{})
 	for _, node := range internalNodes {
 		if !eligibleEpochIdentities.Exists(node.Identity()) {
-			log.Warn().Msgf("node with ID %s is not part of the network according to the bootstrapping data; we might not get any data", node.NodeID)
+			log.Warn().Msgf("Internal node (role=%s id=%x addr=%s) found in internal directory but does not exist in most recent protocol state snapshot. This node will be excluded from recovery epoch.", node.Role, node.NodeID, node.Address)
+			continue
 		}
+		// only add nodes which exist in protocol state snapshot
 		internalNodesMap[node.NodeID] = struct{}{}
 	}
+	// Filter internalNodes so it only contains nodes which are valid for inclusion in the epoch
+	// This is a safety measure: just in case subsequent functions don't properly account for additional nodes,
+	// we proactively remove them from consideration here.
+	internalNodes = slices.DeleteFunc(internalNodes, func(info model.NodeInfo) bool {
+		_, isCurrentEligibleEpochParticipant := internalNodesMap[info.NodeID]
+		return !isCurrentEligibleEpochParticipant
+	})
```

Comment:
My understanding is that this code will modify the slice in-place. I think it is generally a good practise to never modify external inputs, unless they are explicitly designated as such. Even for this simple bootstrapping code, I would suggest to avoid this. 

```suggestion
	{
		temp := make([]bootstrap.NodeInfo, len(internalNodes))
		copy(temp, internalNodes)
		internalNodes = slices.DeleteFunc(temp, func(info model.NodeInfo) bool {
			_, isCurrentEligibleEpochParticipant := internalNodesMap[info.NodeID]
			return !isCurrentEligibleEpochParticipant
		})
		internalNodes = temp
	}
```

Reply:
https://github.com/onflow/flow-go/pull/7127/commits/4a9612caee7e7983803e4e038ecf14a70beb9be7